.travis.yml

```diff
@@ -3,6 +3,15 @@ node_js:
   - '8'
   - '6'
   - '4'
+matrix:
+  include:
+    - node_js: 8
+      env: INTEGRATION=true
+  allow_failures:
+    - node_js: 8
+      env: INTEGRATION=true
 before_install:
   - npm i -g npm
+script:
+  - if [ "$INTEGRATION" = true ]; then npm run integration else npm test; fi
```

Comment:
Ah, you have to quote it:

```
  - "[ \"$INTEGRATION\" = true ] && npm run integration || npm test"
```

Another reason I hate YAML. You never know when you have to quote or not.